File: tdd-katas/vending-machine/vending-machine-javascript/src/vending_machine.js
 export const Choice = {
  COLA: "Cola choice",
  BEER: "Beer choice",
  FIZZY_ORANGE: "Fizzy orange choice"
}

export const Can = {
  NOTHING: "Nothing",
  COKE: "Can of coke",
  FANTA: "Can of fanta"
}

export class VendingMachine {
  constructor() {
    this.choiceCanMap = new Map()
    this.choicePriceMap = new Map()
    this.balanceInCents = 0
  }

  provision(choice, can, priceInCents = 0) {
    this.choiceCanMap.set(choice, can)
    this.choicePriceMap.set(choice, priceInCents)
  }

  insertCoins(amountInCents) {
    this.balanceInCents = amountInCents
  }
  
  deliver(choice) {
    if (!this.choiceCanMap.has(choice)) return Can.NOTHING
    var priceOfCan = this.choicePriceMap.get(choice) 
    
    if (priceOfCan > this.balanceInCents) return Can.NOTHING
    this.balanceInCents -= priceOfCan
    return this.choiceCanMap.get(choice)
  }
}

File: kata-solutions/vending-machine/vending-machine-dotnet/VendingMachine/VendingMachine.cs
using System;
using System.ComponentModel;

namespace Kata;

public enum Can : ushort
{
    [Description("No can")] NOTHING,
    [Description("Can of Coke")] COKE
}

public enum Choice : ushort
{
    [Description("Cola choice")] COLA
}

public class VendingMachine
{
    private Can canOfChoice = Can.NOTHING;
  
    public void Configure(Choice choice, Can can) 
    {
        this.canOfChoice = can;
    }
  
    public Can Deliver(Choice choice)
    {
        return this.canOfChoice;
    }
}
